### PR TITLE
Add user service

### PR DESCRIPTION
**Pull request by blacksam07**
## Summary
- Add user service
- Add user structure

## Trello card
[7-registration-api-integration](https://trello.com/c/8o0olNEK/7-registration-api-integration)

## Screenshots

Screen Shot 2020-04-27 at 13 38 10

Screen Shot 2020-04-27 at 13 38 40

### CODE REVIEW

File: src/app/screens/authentication/components/signup/signup.component.ts [hunk 1]
```diff
@@ -10,7 +12,7 @@ import { keysToSnakeCase } from '../../../../helpers/utils/utils';
 export class SignupComponent implements OnInit {
   authForm: FormGroup;
 
-  constructor(private formBuilder: FormBuilder) { }
+  constructor(private formBuilder: FormBuilder, public http: HttpClient, private userService: UserService) { }
```
Comment (betoobernal) [inline src/app/screens/authentication/components/signup/signup.component.ts]: the accessors should be privates.

File: src/app/screens/authentication/components/signup/signup.component.ts [hunk 2]
```diff
@@ -29,7 +31,9 @@ export class SignupComponent implements OnInit {
 
   onSubmit() {
     this.authForm.value.user = keysToSnakeCase(this.authForm.value.user);
-    console.log(this.authForm.value);
+    this.userService
+    .createUser(this.authForm.value.user)
+    .subscribe(data => console.log('sucess!'))
     this.authForm.reset();
```
Comment (betoobernal) [inline src/app/screens/authentication/components/signup/signup.component.ts]: reset the form when the subscription is fulfilled

File: src/app/services/user.service.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import { environment } from '../../environments/environment'
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserService {
+  apiUrl = `${environment.apiUrl}/users`;
+
+  constructor(private http: HttpClient) { }
+
```
Comment (betoobernal) [inline src/app/services/user.service.ts:15]: remove white spaces

File: src/app/services/user.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export class User {
+  id: number;
+  first_name: string;
+  last_name: string;
+  email: string;
+  password: string;
+  password_confirmation: string;
+  locale: string;
+}
```
Comment (betoobernal) [inline src/app/services/user.ts]: add a newline at end

File: src/app/screens/authentication/components/signup/signup.component.ts [hunk 5]
```diff
@@ -10,7 +12,7 @@ import { keysToSnakeCase } from '../../../../helpers/utils/utils';
 export class SignupComponent implements OnInit {
   authForm: FormGroup;
 
-  constructor(private formBuilder: FormBuilder) { }
+  constructor(private formBuilder: FormBuilder, private http: HttpClient, private userService: UserService) { }
```
Comment (CrisVega08) [inline src/app/screens/authentication/components/signup/signup.component.ts]: why HttpClient Here??

**Review by CrisVega08 — CHANGES_REQUESTED** [review]

File: src/app/services/user.service.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    return this.http.post<any>(
+      this.apiUrl, user
+      ).pipe(catchError (error => throwError('User is already registered')))
+  }
+
+  getUser (userId: number): Observable<User> {
+    return this.http.get(
+      `${this.apiUrl}/${userId}`,
+      ).pipe(
+        map((data: User) => data),
```
Comment (betoobernal) [inline src/app/services/user.service.ts]: why are you using this mapping? 🤔

File: src/app/services/user.service.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+import { Observable, throwError } from 'rxjs';
+import { map, catchError } from 'rxjs/operators';
+import { User } from './user';
+import { environment } from '@env/environment'
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserService {
+  apiUrl = `${environment.apiUrl}/users`;
```
Comment (betoobernal) [inline src/app/services/user.service.ts]: should it be private?

**Review by betoobernal — APPROVED** [review]

**Review by CrisVega08 — APPROVED** [review]

**Pull request merged**